From 0f42058f015ce65e0a6ae1cf17329fea579d9ad5 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Wed, 7 Dec 2016 06:57:23 -0500
Subject: [PATCH 201/268] Restore regrewsion-test cleanliness of C ntpdig.

This code will go away soon, but it's good practice to leave it
in a clean state.
---
 ntpdig/main.c            | 12 ++++++------
 ntpdig/networking.c      |  4 ++--
 ntpdig/utilities.c       | 33 ++++++++++++++++++---------------
 ntpdig/utilities.h       |  2 +-
 tests/ntpdig/utilities.c |  2 +-
 5 files changed, 28 insertions(+), 25 deletions(-)

diff --git a/ntpdig/main.c b/ntpdig/main.c
index c86931e..92aed64 100644
--- a/ntpdig/main.c
+++ b/ntpdig/main.c
@@ -1512,27 +1512,27 @@ offset_calculation(
 	if (debug > 3) {
 		double ftime;
 		struct timespec ts_tmp;
-		pkt_output(rpkt, rpktl, stdout);
+		pkt_output(rpkt, rpktl, false, stdout);
 		printf("ntpdig rootdelay: %f\n", FPTOD(p_rdly));
 		printf("ntpdig rootdisp: %f\n", FPTOD(p_rdsp));
 		printf("ntpdig syncdist: %f\n", *synch_distance);
 		ts_tmp = lfp_stamp_to_tspec(p_ref, NULL);
-		ftime = ts_tmp.tv_sec + ts_tmp.tv_nsec / 1000000.0;
+		ftime = ts_tmp.tv_sec + ts_tmp.tv_nsec / 1e9;
 		printf("ntpdig offset_calculation: ref: %f ", ftime);
 		l_fp_output(&p_ref, stdout);
 		ts_tmp = lfp_stamp_to_tspec(p_org, NULL);
-		ftime = ts_tmp.tv_sec + ts_tmp.tv_nsec / 1000000.0;
+		ftime = ts_tmp.tv_sec + ts_tmp.tv_nsec / 1e9;
 		printf("ntpdig offset_calculation: org: %f ", ftime);
 		l_fp_output(&p_org, stdout);
 		ts_tmp = lfp_stamp_to_tspec(p_rec, NULL);
-		ftime = ts_tmp.tv_sec + ts_tmp.tv_nsec / 1000000.0;
+		ftime = ts_tmp.tv_sec + ts_tmp.tv_nsec / 1e9;
 		printf("ntpdig offset_calculation: rec: %f ", ftime);
 		l_fp_output(&p_rec, stdout);
 		ts_tmp = lfp_stamp_to_tspec(p_xmt, NULL);
-		ftime = ts_tmp.tv_sec + ts_tmp.tv_nsec / 1000000.0;
+		ftime = ts_tmp.tv_sec + ts_tmp.tv_nsec / 1e9;
 		printf("ntpdig offset_calculation: xmt: %f ", ftime);
 		l_fp_output(&p_xmt, stdout);
-		ftime = tv_dst->tv_sec + tv_dst->tv_usec / 1000.0;
+		ftime = tv_dst->tv_sec + tv_dst->tv_usec / 1e6;
 		printf("ntpdig dst %f\n", ftime);
 	}
 #endif
diff --git a/ntpdig/networking.c b/ntpdig/networking.c
index 8162b1b..1143058 100644
--- a/ntpdig/networking.c
+++ b/ntpdig/networking.c
@@ -17,7 +17,7 @@ sendpkt (
 #ifdef DEBUG
 	if (debug > 2) {
 		printf("ntpdig sendpkt: Packet data:\n");
-		pkt_output(pkt, len, stdout);
+		pkt_output(pkt, len, true, stdout);
 	}
 #endif
 	TRACE(1, ("ntpdig sendpkt: Sending packet to %s ...\n",
@@ -56,7 +56,7 @@ recvdata(
 #ifdef DEBUG
 	if (debug > 2) {
 		printf("Received %d bytes from %s:\n", recvc, sockporttoa(sender));
-		pkt_output((struct pkt *)rdata, recvc, stdout);
+		pkt_output((struct pkt *)rdata, recvc, true, stdout);
 	}
 #endif
 	return recvc;
diff --git a/ntpdig/utilities.c b/ntpdig/utilities.c
index 077c191..e470ab4 100644
--- a/ntpdig/utilities.c
+++ b/ntpdig/utilities.c
@@ -8,7 +8,8 @@
 void 
 pkt_output (
 		struct pkt *dpkg,
-		int pkt_length, 
+		int pkt_length,
+		bool longform,
 		FILE *output
 	   )
 {
@@ -17,24 +18,26 @@ pkt_output (
 
 	pkt = (uint8_t *)dpkg;
 
-#ifdef __UNUSED__
-	fprintf(output, HLINE);
-
-	for (a = 0; a < pkt_length; a++) {
-		if (a > 0 && a % 8 == 0)
-			fprintf(output, "\n");
+	if (longform) {
+	    fprintf(output, HLINE);
 
-		fprintf(output, "%d: %x \t", a, pkt[a]);
-	}
+	    for (a = 0; a < pkt_length; a++) {
+		    if (a > 0 && a % 8 == 0)
+		        fprintf(output, "\n");
 
-	fprintf(output, "\n");
-	fprintf(output, HLINE);
-#endif /* __UNUSED__ */
+		    fprintf(output, "%d: %x \t", a, pkt[a]);
+	    }
 
-	for (a = 0; a < pkt_length; a++) {
-	    fprintf(output, "%02x", pkt[a]);
+	    fprintf(output, "\n");
+	    fprintf(output, HLINE);
+	}
+	else
+	{
+		for (a = 0; a < pkt_length; a++) {
+		    fprintf(output, "%02x", pkt[a]);
+		}
+		fprintf(output, "\n");
 	}
-	fprintf(output, "\n");
 
 }
 
diff --git a/ntpdig/utilities.h b/ntpdig/utilities.h
index 772f370..99877c5 100644
--- a/ntpdig/utilities.h
+++ b/ntpdig/utilities.h
@@ -14,7 +14,7 @@
 #define STDLINE printf(HLINE);
 
 
-void pkt_output(struct pkt *dpkg, int pkt_length, FILE *output);
+void pkt_output(struct pkt *dpkg, int pkt_length, bool longform, FILE *output);
 void l_fp_output(l_fp *ts, FILE *output);
 void l_fp_output_bin(l_fp *ts, FILE *output);
 void l_fp_output_dec(l_fp *ts, FILE *output);
diff --git a/tests/ntpdig/utilities.c b/tests/ntpdig/utilities.c
index 85979e3..131ccd8 100644
--- a/tests/ntpdig/utilities.c
+++ b/tests/ntpdig/utilities.c
@@ -154,7 +154,7 @@ TEST(utilities, DebugPktOutput) {
 	test.l_uf = 2147483647; // Lots of ones.
 	HTONL_FP(&test, &testpkt.xmt);
 
-	pkt_output(&testpkt, LEN_PKT_NOMAC, outputFile);
+	pkt_output(&testpkt, LEN_PKT_NOMAC, true, outputFile);
 
 	FinishDebugTest(CreatePath("debug-input-pkt", INPUT_DIR), filename);
 }
-- 
2.7.4

